drivers/clock_control/clock_control_nrf_auxpll.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_auxpll
+
+#include <stdint.h>
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/toolchain.h>
+
+#include <hal/nrf_auxpll.h>
+
+/* obtain range value from configuration */
+#define AUXPLL_RANGE_FROM_CONFIG(val) (FIELD_GET(AUXPLL_CONFIG_CFGSTATIC_AUXPLLRANGE_Msk, val) + 3U)
+
+struct clock_control_nrf_auxpll_config {
+	NRF_AUXPLL_Type *auxpll;
+	uint32_t ref_clk_hz;
+	uint32_t ficr_ctune;
+	uint32_t cfg;
+	uint16_t frequency;
+	nrf_auxpll_ctrl_outsel_t out_div;
+};
+
+static int clock_control_nrf_auxpll_on(const struct device *dev, clock_control_subsys_t sys)
+{
+	const struct clock_control_nrf_auxpll_config *config = dev->config;
+	uint32_t val;
+
+	ARG_UNUSED(sys);
+
+	nrf_auxpll_task_trigger(config->auxpll, NRF_AUXPLL_TASK_START);
+
+	do {
+		val = nrf_auxpll_status_get(config->auxpll) & AUXPLL_STATUS_MODE_Msk;
+	} while (val != (AUXPLL_STATUS_MODE_Locked << AUXPLL_STATUS_MODE_Pos));
+
+	return 0;
+}
+
+static int clock_control_nrf_auxpll_off(const struct device *dev, clock_control_subsys_t sys)
+{
+	const struct clock_control_nrf_auxpll_config *config = dev->config;
+	uint32_t val;
+
+	ARG_UNUSED(sys);
+
+	nrf_auxpll_task_trigger(config->auxpll, NRF_AUXPLL_TASK_STOP);
+
+	do {
+		val = nrf_auxpll_status_get(config->auxpll) & AUXPLL_STATUS_PLLRUNNING_Msk;
+	} while (val != (AUXPLL_STATUS_PLLRUNNING_Off << AUXPLL_STATUS_PLLRUNNING_Pos));
+
+	return 0;
+}
+
+static int clock_control_nrf_auxpll_get_rate(const struct device *dev, clock_control_subsys_t sys,
+					     uint32_t *rate)
+{
+	const struct clock_control_nrf_auxpll_config *config = dev->config;
+	uint8_t range;
+
+	ARG_UNUSED(sys);
+
+	range = AUXPLL_RANGE_FROM_CONFIG(config->auxpll->CONFIG.CFGSTATIC);
```

Comment:
```suggestion
	nrf_auxpll_config_t cfg;
	nrf_auxpll_config_get(config->auxpll, &cfg);
```
Then you can do `cfg.range + 3U`. I'll look into replacing this magic number with a define.